### PR TITLE
Version 2.0

### PR DESCRIPTION
**Pull request by FireInstall**
- builds on top of addstars (Narimm) 1.0 version
- Refactor code to clean up!

### CODE REVIEW

**Comment by FireInstall** [comment]
@addstar I hope it's ok that I used [the jar file on your maven](https://maven.addstar.com.au/artifactory/ext-snapshot-local/org/kitteh/paste-gg-api/1.0.0-SNAPSHOT/) to copy the api-key and deletion code.
I'm happy to open a PR on your Repo as well, if you make it public.

(until we have an awnser this should not get merged!)

**Comment by addstar** [comment]
> @addstar I hope it's ok that I used [the jar file on your maven](https://maven.addstar.com.au/artifactory/ext-snapshot-local/org/kitteh/paste-gg-api/1.0.0-SNAPSHOT/) to copy the api-key and deletion code.

Heya @FireInstall :wave: I don't even remember what this is for or which key you mean?

> I'm happy to open a PR on your Repo as well, if you make it public.

Which repo do you want made public? We're happy for you to contribute to any of our stuff :slightly_smiling_face: you make a lot of great changes!

**Comment by FireInstall** [comment]
@addstar 
This is the paste.gg api made by  Matt Baxter, who made https://kitteh.org (dead now)
The Minigames Plugin depends on it, since the debug commands collects much data and posts it on [paste.gg](https://paste.gg/) - a paste bin. You can add your files anonymously and it works fine.
But Minigames uses a feature the [original public code](https://github.com/KittehOrg/PasteGGAPI) doesn't include: the use of api - keys for paste.gg. It helps to keep track of pastes past the next server restart.
Narimm contributed in the past to the project and the specific .jar Minigames uses stems from [your maven](https://maven.addstar.com.au/artifactory/ext-snapshot-local/org/kitteh/paste-gg-api/1.0.0-SNAPSHOT/). 

Well, seems like I didn't look close enough. GitHub didn't show me the list of forks of the original repo, so I assumed it's a private one of your organisation.
However, seems like Narimm just created it under [his account](https://github.com/Narimm/PasteGGAPI/tree/master-mb). Maybe he did upload it to your maven manually? I don't know.
Minigames doesn't really profit from most of my work here - but it should receive dependency upgrades at least, since Minigames did run far too long with a security issue based on a Gson attack.


tl;dr: Since you inherited Narimms legacy:

1. Do you approve of the usage of his code here? (should be all under MIT license anyway)
2. We need a new home for the code and a maven repo. Do you want to keep it all under your umbrella and set everything up (if missing) and I open a PR there or do you want us to host it in the future?
